migration/migrator_test.go

```diff
@@ -0,0 +1,281 @@
+package migration
+
+import (
+	"fmt"
+	"math/big"
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/consensus/ethash"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/core/rawdb"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/core/vm"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/ethdb"
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/ethereum/go-ethereum/trie"
+	"github.com/ethereum/go-ethereum/trie/trienode"
+)
+
+func TestApplyNewStateTransition(t *testing.T) {
+	genesis := &core.Genesis{
+		Config: params.KromaTestConfig,
+	}
+	chainDB := rawdb.NewMemoryDatabase()
+	cacheConfig := core.DefaultCacheConfigWithScheme(rawdb.HashScheme)
+	cacheConfig.Preimages = true
+	cacheConfig.KromaZKTrie = true
+	blockchain, _ := core.NewBlockChain(chainDB, cacheConfig, genesis, nil, ethash.NewFaker(), vm.Config{}, nil, nil)
+	blockchain.TrieDB().SetBackend(true)
+
+	eth := fakeEthBackend{
+		chainDB, blockchain,
+	}
+
```

Comment:
Add blocks including transactions here

Reply:
I added it below because addBlock() require the migrator pointer